templates/common/on-prem/files/NetworkManager-resolv-prepender.yaml

```diff
@@ -52,14 +52,21 @@ contents:
             "{{ onPremPlatformIngressIP . }}")
         DOMAIN="{{.DNS.Spec.BaseDomain}}"
         if [[ -n "$NAMESERVER_IP" ]]; then
-            >&2 echo "NM resolv-prepender: Prepending 'nameserver $NAMESERVER_IP' to /etc/resolv.conf (other nameservers from /var/run/NetworkManager/resolv.conf)"
-            sed -e "/^search/d" \
-                -e "/Generated by/c# Generated by KNI resolv prepender NM dispatcher script\nsearch $DOMAIN\nnameserver $NAMESERVER_IP" \
-                /var/run/NetworkManager/resolv.conf > /etc/resolv.tmp
+            source /etc/os-release
+            if [[ "$NAME" == "Fedora" ]]; then
```

Comment:
What sets the $NAME variable? Don't you also need to check the version of Fedora?

Perhaps a better check would be to look for the use of systemd-resolved, so that there's no surprise when RHCOS switches to it.

Reply:
> What sets the $NAME variable?

Its being set after `source /etc/os-release`

> Don't you also need to check the version of Fedora?

We could add this later, but OKD 4.6 uses F33+, so systemd-resolved would certainly be used

> Perhaps a better check would be to look for the use of systemd-resolved

That was my initial approach, but its not clear if there is a nice way to do that - check that /etc/resolv.conf is a symlink? systemd-resolved service is running?

Reply:
I think the most robust is `systemctl is-active --quiet systemd-resolved`

Reply:
For some reason that didn't work. I'll have another look and follow up on that later

Reply:
Yeah, don't worry too much about it. I think we'll know where to look if suddenly DNS stops working.

Reply:
I think checking for the link should be fine.